pkg/controllers/jobset_controller.go

```diff
@@ -916,3 +939,98 @@ func findReplicatedStatus(replicatedJobStatus []jobset.ReplicatedJobStatus, repl
 	}
 	return jobset.ReplicatedJobStatus{}
 }
+
+func (r *JobSetReconciler) deleteJobSet(ctx context.Context, js *jobset.JobSet) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	policy := metav1.DeletePropagationForeground
+	options := []client.DeleteOption{client.PropagationPolicy(policy)}
+	log.V(2).Info("Cleaning up JobSet", "jobset", klog.KObj(js))
+
+	return r.Delete(ctx, js, options...)
+}
+
+// checkIfTTLExpired checks whether a given JobSet's TTL has expired.
+func (r *JobSetReconciler) checkIfTTLExpired(ctx context.Context, jobSet *jobset.JobSet) (*time.Time, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// We don't care about the JobSets that are going to be deleted
+	if jobSet.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	now := r.clock.Now()
+	remaining, expiredAt, err := timeLeft(log, jobSet, &now)
+	if err != nil {
+		return nil, err
+	}
+
+	// TTL has expired
+	if remaining != nil && *remaining <= 0 {
+		return expiredAt, nil
+	}
+
+	return nil, nil
+}
+
+// timeLeft returns the time left until the JobSet's TTL expires and the time when it will expire.
+func timeLeft(log logr.Logger, js *jobset.JobSet, since *time.Time) (*time.Duration, *time.Time, error) {
```

Comment:
Can we rename `since` to something like `now` or `currTime` since it holds the current time? The variable name `since` makes the code a bit more confusing for me to read